src/components/KToaster/ToastManager.ts

```diff
@@ -1,87 +1,83 @@
 import { ToasterAppearances } from '@/types'
+import KToaster from '@/components/KToaster/KToaster.vue'
+import { v4 as uuidv4 } from 'uuid'
 
-const APPEARANCES = Object.keys(ToasterAppearances)
+const toasterContainerId = 'kongponents-toaster-container'
 
-const DEFAULTS = {
-  id: 'toaster-container',
-  timeout: 5000,
+const toasterDefaults = {
+  timeoutMilliseconds: 5000,
   appearance: ToasterAppearances.info,
-  zIndex: 10000,
 }
 
-export default class ToastManager {
-  public toasters: Ref<Toast[]>
-  public timeout: number
-  public appearance: string
-  public id: string
-  public zIndex: number
-
-  constructor({ id, timeout, appearance, zIndex } = DEFAULTS) {
-    this.toasters = ref<Toast[]>([])
+const defaultZIndex = 10000
 
-    this.timeout = timeout
-    this.appearance = appearance
-    this.id = id
-    this.zIndex = zIndex
-
-    this.mount()
-  }
+export default class ToastManager {
+  private toastersContainer: HTMLElement | null = null
+  private toaster: VNode | null = null
+  public toasts: Ref<Toast[]> = ref<Toast[]>([])
 
-  mount() {
-    // For SSR (still works on VuePress/VitePress site, but prevents failing on the build)
+  constructor(options?: ToasterOptions) {
+    // For SSR, prevents failing on the build)
     if (typeof document === 'undefined') return
 
-    // Create a component container for the notification to bind to
-    const notificationContainer = document.createElement('div')
-    notificationContainer.id = this.id
-    document.body.appendChild(notificationContainer)
+    this.toastersContainer = document.createElement('div')
+    this.toastersContainer.id = toasterContainerId
+    document.body.appendChild(this.toastersContainer)
 
-    const Toast = h(KToaster, {
-      toasterState: this.toasters.value,
-      zIndex: this.zIndex,
+    this.toaster = createVNode(KToaster, {
+      toasterState: this.toasts.value,
+      zIndex: options?.zIndex ? options.zIndex : defaultZIndex,
       onClose: (key: any) => this.close(key),
     })
 
-    createApp(Toast).mount(`#${this.id}`)
+    if (this.toastersContainer) {
+      render(this.toaster, this.toastersContainer)
+    }
   }
 
   setTimer(key: any, timeout: number): number {
     return setTimeout(() => this.close(key), timeout)
   }
 
-  open(args: Record<string, any> | string): void {
+  public open(args: Record<string, any> | string): void {
     // eslint-disable-next-line @typescript-eslint/ban-ts-comment
     // @ts-ignore
     const { key, timeoutMilliseconds, appearance, message, title } = args
 
-    const _key: any = key || (this.toasters.value.length) + new Date().getTime()
-    const _appearance: ToasterAppearance = (appearance && APPEARANCES.indexOf(appearance) !== -1) ? appearance : this.appearance
-    const timer: number = this.setTimer(_key, timeoutMilliseconds || this.timeout)
-    const _message = typeof args === 'string' ? args : message
+    const toastKey: any = key || uuidv4()
```

Comment:
uuid again (just pointing it out so we don't forget, I know you need the other PR ❤️ )

Reply:
Updated